db/functions.js

```diff
@@ -0,0 +1,15 @@
+const db = require( './config' ).db
+const pgp = require( './config' ).pgp
+
+const getNextStation = function(current_station){
+  return db.any(`SELECT station_number, (SELECT COUNT(*) FROM stations) FROM stations WHERE station_name = $1 GROUP BY station_number`, current_station)
```

Comment:
all you need is return the station number and increment by one and return all from that. this function looks like it is being extra specific when it doesn't need to be.